jemalloc-sys/build.rs

```diff
@@ -28,14 +28,32 @@ macro_rules! warning {
     }
 }
 
+// Stolen from openssl-sys
+fn env_inner_os(name: &str) -> Option<OsString> {
```

Comment:
How about merging these duplicated small functions into one like:
```
fn read_and_watch_env_imp<T, F: Fn(&str) -> T>(name: &str, read: F) -> T {
    let imp = |n| {
        println!("rerun-if-change");
        read(n)
    };
    let prefixed = format!("");
    imp(&prefixed).or_else(|| imp(name))
}
```?

Reply:
I was following the original code structure which splitting env::var and env::var_os, I guess this shall be done by another commit after merging this pr?

Reply:
My proposal still keeps `read_and_watch_env` and `read_and_watch_env_os` as two functions, just to avoid adding more small functions.

Reply:
i see